### PR TITLE
Add time format used by some SDO HMI FITS keywords

### PR DESCRIPTION
**Pull request by Gbear605**
This was found in the SDO HMI Intensitygram file HMI20120606_205622.fits. The T_OBS, T_REC, and TRECEPOC keywords contain a time string with this format.

### CODE REVIEW

**Comment by Gbear605** [comment]
So the travis build failed, but I'm pretty sure all the failures are due to dependencies, not due to my changes.

File: sunpy/time/time.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "%Y:%j:%H:%M:%S",          # Example 2012:124:21:08:12
     "%Y:%j:%H:%M:%S.%f",       # Example 2012:124:21:08:12.999999
     "%Y%m%d%H%M%S",            # Example 20140101000001 (JSOC / VSO)
+    "%Y.%m.%d_%H:%M:%S_TAI",   # Example 2016.05.04_21:08:12_TAI
```
Comment (Cadair) [inline sunpy/time/time.py]: I am happy to add this here, but you should know we assume all times are UTC so this may cause unexpected accuracy issues.

**Comment by Cadair** [comment]
Thanks for the PR @Gbear605.

As I mentioned above, at the moment, until we support Astropy Time throughout sunpy in other places in the code we are assuming that all times are UTC. Is this likely to cause a problem here?

**Comment by Gbear605** [comment]
I do not believe that this will cause an issue because UTC and TAI will always be within one second of each other due to leap seconds, and the time format added is only accurate to the nearest second.

**Comment by Cadair** [comment]
Cool looks good to me as long as travis sorts itself out. :+1:

**Comment by Gbear605** [comment]
Oh sorry, I had just misunderstood what TAI was. UTC and TAI will be off by some amount of seconds, but for the time being it should not be major since the difference only changes by roughly one second per two years. While this difference is unfortunate, it seems likely to be a better fix than simply ignoring the existence of this date format that is used in HMI data.

Adding a note about the difference between TAI and UTC might be helpful for the documentation.

**Comment by dpshelio** [comment]
:+1:  happy - but we need to create an issue about this test for when `astropy.time` comes in.

**Comment by Cadair** [comment]
build fail is unrelated.

**Pull request merged**